src/commands/createNewProject/PythonProjectCreator.ts

```diff
@@ -0,0 +1,170 @@
+                    },
+                    linux: {
+                        command: `source ${path.join('.', '\${config:azureFunctions.python.venvRelativePath}', 'bin', 'activate')} && func start host`
+                    },
+                    isBackground: true,
+                    presentation: {
+                        reveal: 'always'
+                    },
+                    options: {
+                        env: {
+                            // tslint:disable-next-line:no-invalid-template-strings
+                            'languageWorkers:python:arguments': '-m ptvsd --server --port 9091 --file'
+                        }
+                    },
+                    problemMatcher: [
+                        funcHostProblemMatcher
+                    ]
+                }
+            ]
+        };
+    }
+
+    private async validatePythonVersion(): Promise<void> {
+        const minReqVersion: string = '3.6.0';
+        await this.tryGetPythonAlias(PythonAlias.py, minReqVersion);
+        await this.tryGetPythonAlias(PythonAlias.python, minReqVersion);
+        await this.tryGetPythonAlias(PythonAlias.python3, minReqVersion);
+        if (!this.pythonAlias) {
+            throw new Error(localize('pythonVersionRequired', 'Python {0} is required to create a Python Function project with Azure Functions Core Tools.', minReqVersion));
+        }
+    }
+
+    private async tryGetPythonAlias(pyAlias: PythonAlias, minReqVersion: string): Promise<void> {
+        try {
+            const pyVersion: string = (await cpUtils.executeCommand(ext.outputChannel, undefined /*default to cwd*/, `${pyAlias} --version`)).substring('Python '.length);
+            if (semver.gte(pyVersion, minReqVersion)) {
+                this.pythonAlias = pyAlias;
+            }
+        } catch {
+            // ignore error, not installed under this alias
+        }
+    }
+
+    private async setVirtualEnviornment(): Promise<void> {
+        const funcEnv: string = 'func_env';
+        let pythonVenvPath: string;
+        const newButton: MessageItem = { title: 'Create New' };
+        const existingButton: MessageItem = { title: 'Use Existing' };
+        const venvRequired: string = localize('venvRequired', 'You must be running in a Python virtual environment to create a Python Function project. Create a new one or use an existing?');
+        const input: MessageItem = await ext.ui.showWarningMessage(venvRequired, { modal: true }, newButton, existingButton);
+        if (input === newButton) {
+            // if there is no func_env, create one as it's required for Python functions
+            await cpUtils.executeCommand(ext.outputChannel, this.functionAppPath, this.pythonAlias, '-m', 'venv', 'func_env');
+            pythonVenvPath = funcEnv;
+        } else {
+            const options: OpenDialogOptions = {
```

Comment:
Should this still be a browse dialog or just an input box where they enter the name of the venv?

If we had an input box, I feel like it might say this:
> Enter the name of the Python virtual environment required for this project. It will be created if it does not already exist.

You might not even need the warning above if you do this. And obviously the input box should default to a value of 'func_env'

Reply:
Well, I'm not sure how Python users use virtual environments, but the folder that contains the venv doesn't need to be within the same root folder of the project.

If we give someone the option to use their own venv, I'd imagine it wouldn't be within their Python project (since they are creating a new project at this moment).

I don't know how common it would be, but I guess someone could link it to an existing venv that is already set up to use Python Functions.

Reply:
Okay well I think the existing 'browse' option is confusing (it's not clear if I'm supposed to select the folder containing the 'activate' command or the root folder of the venv). I'm also not sure what the use case is for an existing venv (especially one outside your repo). I would rather simplify our user experience than cater to an edge-case that we don't understand. Plus those edge-case users have a workaround (manually edit their tasks.json)

What if we remove all venvRelativePath stuff from this PR and create a separate issue to debate/discuss/address the "existing venv" scenario you're talking about?

Reply:
(But obviously keep a branch with your existing venvRelativePath stuff just in case we end up going with it)

Reply:
Mmkay.  I'm okay with that.